pkg/metadata/prometheus_test.go

```diff
@@ -36,17 +41,58 @@ scrape_configs:
   scrape_interval: 1s
   scrape_timeout: 1s
   static_configs:
-  - targets: ['localhost:9090','localhost:80']
+  - targets: ['localhost:9090']
 `))
 	testutil.Ok(t, p.Start())
 
-	time.Sleep(10 * time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	_ = ctx
+
+	upctx, upcancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer upcancel()
+
+	logger := log.NewNopLogger()
+	// Wait for Prometheus up.
+	err = p.WaitPrometheusUp(upctx, logger)
+	testutil.Ok(t, err)
+
+	// Update the scrape configs to the current Prometheus address.
+	err = p.SetConfig(fmt.Sprintf(`
+global:
+  external_labels:
+    region: eu-west
+scrape_configs:
+- job_name: 'myself'
+  # Quick scrapes for test purposes.
+  scrape_interval: 1s
+  scrape_timeout: 1s
+  static_configs:
+  - targets: ['%s']
+`, p.Addr()))
+	testutil.Ok(t, err)
+
+	// Do a hot reload for loading the latest configuration.
+	err = p.Reload()
+	testutil.Ok(t, err)
 
 	u, err := url.Parse("http://" + p.Addr())
 	testutil.Ok(t, err)
 
-	prom := NewPrometheus(u, promclient.NewDefaultClient())
+	c := promclient.NewClient(http.DefaultClient, logger, "")
```

Comment:
It is used in `meta, err := c.MetadataInGRPC(ctx, u, "", -1)`.

Reply:
I think what @bwplotka meant is that we only use it in one place so it can be in-lined but I'd argue that it's more readable the way it is right now because you do not have a chain of function calls with quite a few arguments

Reply:
Not just that but also `grpcClient := NewGRPCClient(NewPrometheus(u, c))` so it makes sense so leave this as is. Plus, it's more readable to have it this way instead of chaining function calls with quite a few parameters.